checker/src/main/java/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -1,1035 +1,145 @@
-                assert p instanceof AnnotatedExecutableType;
-                AnnotatedExecutableType exeType = (AnnotatedExecutableType) p;
-                DeclaredType underlyingType =
-                        (DeclaredType) exeType.getReturnType().getUnderlyingType();
-                AnnotationMirror a = getUnderInitializationAnnotationOfSuperType(underlyingType);
-                exeType.getReturnType().replaceAnnotation(a);
-            }
-            return result;
-        }
-
-        @Override
-        public Void visitNewClass(NewClassTree tree, AnnotatedTypeMirror p) {
-            super.visitNewClass(tree, p);
-            boolean allInitialized = true;
-            Type type = ((JCTree) tree).type;
-            for (ExpressionTree a : tree.getArguments()) {
-                final AnnotatedTypeMirror t = getAnnotatedType(a);
-                allInitialized &= (isInitialized(t) || isFbcBottom(t));
-            }
-            if (!allInitialized) {
-                p.replaceAnnotation(createUnderInitializationAnnotation(type));
-                return null;
-            }
-            p.replaceAnnotation(INITIALIZED);
-            return null;
-        }
-
-        @Override
-        public Void visitLiteral(LiteralTree tree, AnnotatedTypeMirror type) {
-            if (tree.getKind() != Tree.Kind.NULL_LITERAL) {
-                type.addAnnotation(INITIALIZED);
-            }
-            return super.visitLiteral(tree, type);
-        }
-
-        @Override
-        public Void visitMemberSelect(
-                MemberSelectTree tree, AnnotatedTypeMirror annotatedTypeMirror) {
-            if (TreeUtils.isArrayLengthAccess(tree)) {
-                annotatedTypeMirror.replaceAnnotation(INITIALIZED);
-            }
-            return super.visitMemberSelect(tree, annotatedTypeMirror);
-        }
+    public List<Pair<ReturnNode, TransferResult<CFValue, InitializationStore>>>
+            getReturnStatementStores(MethodTree methodTree) {
+        return getFieldAccessFactory().getReturnStatementStores(methodTree);
     }
 
     /**
-     * The {@link QualifierHierarchy} for the initialization type system.
+     * Determines whether the specified variable's current value is initialized.
      *
-     * <p>Type systems extending the Initialization Checker should call methods {@link
-     * InitializationQualifierHierarchy#isSubtypeInitialization} and {@link
-     * InitializationQualifierHierarchy#leastUpperBoundInitialization} for appropriate qualifiers.
-     * See protected subclass NullnessQualifierHierarchy within class {@link NullnessChecker} for an
-     * example.
+     * <p>Returns {@code true} iff the variable's current value is initialized. This holds for
+     * variables whose value has a non-top qualifier that does not have the meta-annotation {@link
+     * HoldsForDefaultValues} as well as variables whose declaration has a qualifier that {@link
+     * HoldsForDefaultValues}.
```

Comment:
Can you try rewording this to make it clearer?
In particular, what does `variables whose declaration has a qualifier that HoldsForDefaultValues` mean? That the declared type of the variable has that meta-annotation?